From 058c5007d316195a2a2f543918fc7650cdb68135 Mon Sep 17 00:00:00 2001
From: Wayne Zou <b36644@freescale.com>
Date: Fri, 10 Jun 2011 17:02:23 +0800
Subject: [PATCH] ENGR00144479 MX53_SMD Suspend/Resume: Auto resume from stop mode at first time

The GPIO9 on da9053 chip is floating on board, and it is configured as input
before suspend. Fixed it by mask the interrupt source on da9053.

Signed-off-by: Wayne Zou <b36644@freescale.com>
---
 arch/arm/mach-mx5/pm_da9053.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx5/pm_da9053.c b/arch/arm/mach-mx5/pm_da9053.c
index eb48e33..b75cdda 100644
--- a/arch/arm/mach-mx5/pm_da9053.c
+++ b/arch/arm/mach-mx5/pm_da9053.c
@@ -341,7 +341,8 @@ void pm_da9053_dump(int start, int end)
 
 #define DA9052_CONTROLC_SMD_SET 0x62
 #define DA9052_GPIO0809_SMD_SET 0x18
-#define DA9052_ID1415_SMD_SET 0x1
+#define DA9052_ID1415_SMD_SET   0x1
+#define DA9052_GPI9_IRQ_MASK    0x2
 
 int da9053_suspend_cmd_sw(void)
 {
@@ -399,6 +400,9 @@ int da9053_suspend_cmd_hw(void)
 
 	pm_da9053_write_reg(DA9052_GPIO0809_REG,
 			DA9052_GPIO0809_SMD_SET);
+	pm_da9053_read_reg(DA9052_IRQMASKD_REG, &data);
+	data |= DA9052_GPI9_IRQ_MASK;
+	pm_da9053_write_reg(DA9052_IRQMASKD_REG, data);
 
 	pm_da9053_read_reg(DA9052_ID1415_REG, &data);
 	data &= 0xf0;
-- 
1.5.4.4

